src/statespace/SE2.cpp

```diff
@@ -53,6 +50,18 @@ Eigen::VectorXd SE2::State::vector() const {
     return state_vector;
 }
 
+bool SE2::State::from_vector(const Eigen::VectorXd& vector) {
+    if (vector.size() != 3) {
+        return false;
+    }
```

Comment:
We should probably throw an exception instead of returning false
```
if (state.rows() != 3) {
    std::stringstream msg;
    msg << "state has incorrect size: expected 3"
        << ", got " << state.size() << ".\n";
    throw std::runtime_error(msg.str());
  }
```